src/super_gradients/training/models/segmentation_models/unet/unet_encoder.py

```diff
@@ -0,0 +1,268 @@
+        """
+        STDC block refactors the convolution operator by applying several smaller convolution with num of filters that
+        decrease w.r.t the num of steps. The ratio to the smallest num of channels is `2 ** (steps - 1)`,
+        thus this method assert that the stage num of channels is divisible by the above ratio.
+        """
+        channels_ratio = 2 ** (steps - 1)
+        if num_channels % channels_ratio != 0:
+            raise AssertionError(f"Num channels: {num_channels}, isn't divisible by the channels width ratio:"
+                                 f" {channels_ratio}, when initiating an STDC block with steps: {steps}")
+
+
+class RepVGGStage(BackboneStage):
+    """
+    RepVGG stage with RepVGGBlock as building block. If `anti_alias=True`, `AntiAliasDownsample` module is used for
+    downsampling.
+    """
+    def build_stage(self, in_channels: int, out_channels: int, stride: int, num_blocks: int, anti_alias: bool,
+                    **kwargs):
+        blocks = []
+        # Anti alias gaussian down-sampling
+        if anti_alias and stride == 2:
+            blocks.append(AntiAliasDownsample(in_channels, stride))
+            stride = 1
+        # RepVGG blocks
+        blocks.extend([
+            RepVGGBlock(in_channels, out_channels, kernel_size=3, stride=stride, padding=1),
+            *[RepVGGBlock(out_channels, out_channels, kernel_size=3, padding=1) for _ in range(num_blocks - 1)]
+        ])
+        return nn.Sequential(*blocks)
+
+
+class RegnetXStage(BackboneStage):
+    """
+    RegNetX stage with XBlock as building block.
+    """
+    def build_stage(self, in_channels: int, out_channels: int, stride: int, num_blocks: int, bottleneck_ratio: float,
+                    group_width: int, se_ratio: float, droppath_prob: float, **kwargs):
+        group_width = self._get_divisable_group_width(out_channels, bottleneck_ratio, group_width)
+        return nn.Sequential(
+            XBlock(in_channels, out_channels, bottleneck_ratio, group_width, stride, se_ratio, droppath_prob),
+            *[XBlock(out_channels, out_channels, bottleneck_ratio, group_width, 1, se_ratio, droppath_prob)
+              for _ in range(num_blocks - 1)]
+        )
+
+    @staticmethod
+    def _get_divisable_group_width(channels: int, bottleneck_ratio: float, group_width: int) -> int:
+        """
+        Returns a valid value for group_width, in channels isn't a multiplication of group_width.
+        """
+        inter_channels = channels // bottleneck_ratio
+        # if group_width is higher than the Conv channels, fallback to a regular Conv with group_width = channels.
+        if group_width > inter_channels:
+            return inter_channels
+        group_pow = int(math.log2(group_width))
+        for pow in range(group_pow, -1, -1):
+            if (inter_channels / 2 ** pow) % 1 == 0:
+                return int(2 ** pow)
+        return 1
+
+
+class BlockType(Enum):
```

Comment:
here too - use factories instead of enums

Reply:
Done